### PR TITLE
Bugfix for specifying relative paths in .aiida-testing-config.yml

### PR DESCRIPTION
**Pull request by janssenhenning**
The path manipulation was using the full path of the config file
instead of the path to the directory of the config file

This was not noticed since the executable is only executed when
regenerating test data and no test was added in aiida-testing (My bad :()
A test is now added

### CODE REVIEW

File: aiida_testing/mock_code/_fixtures.py
Comment (ltalirz) [inline aiida_testing/mock_code/_fixtures.py]: Just wondering what should be the logical order of these different approaches (also on the lines above).

Here is what would make sense to me:

 1. check if it is an absolute path. if yes, take that
 2. check if it is a relative path with respect to the config file. if yes, take that
 3. check if the executable is in the PATH

Would you agree?

**Review by ltalirz — COMMENTED** [review]
thanks for the fix @janssenhenning , 
I just have one small suggestion about the order of how things are checked

File: aiida_testing/mock_code/_fixtures.py
Comment (janssenhenning) [reply aiida_testing/mock_code/_fixtures.py]: Yes I think that is a sensible order 👍 
I have one question about this code part
It starts with 
```python
code_executable_path = mock_code_config.get(label, 'TO_SPECIFY')
if (not code_executable_path) and executable_name:
    code_executable_path = shutil.which(executable_name) or 'NOT_FOUND'
```
If the code is not specified at all it will never take the `executable_name` into account because of the default
So should the executable_name only be relevant if you set the code to an empty string or is this a mistake?

Comment (ltalirz) [reply aiida_testing/mock_code/_fixtures.py]: Hey @janssenhenning , sorry for the delay.

Indeed, that is a bug and it turns out it was introduced by myself - shame on me, also for not adding any test for the `executable_name` feature ;-)

I believe my idea would have been that you could forego the need for the config file simply by passing the name of the executable to the pytest fixture and making sure it was in the PATH when the tests are run.

As you point out, at least as the code stands now that would actually only work when using a config file that explicitly sets the code path to an empty value.

I can't remember ever purposefully doing that - given this, I would also be ok with removing the `executable_name` parameter altogether (I don't think it will break anything).
I leave it up to you whether you want to make it work or remove it.

Comment (ltalirz) [reply aiida_testing/mock_code/_fixtures.py]: Just waiting on reply from @janssenhenning before moving forward

Comment (ltalirz) [reply aiida_testing/mock_code/_fixtures.py]: ping

Comment (janssenhenning) [reply aiida_testing/mock_code/_fixtures.py]: Sorry for being inactive on this. I now adjusted the path determination to this order and added a test for the executable name given case.

I also added two more error messages for the cases, where
- A relative path is given but no executable is found at that path
- no path but the executable name is given and it is not found on the PATH

**Comment by janssenhenning** [comment]
@ltalirz I don't know what to make of the failing tests. Did something change in how the AiiDA test profile has to be set up?

**Comment by ltalirz** [comment]
Thanks @janssenhenning ! I'll have a look into the erorr

**Comment by ltalirz** [comment]
Fixed, can you please rebase on `develop` ?

File: tests/mock_code/test_diff.py
Comment (ltalirz) [inline tests/mock_code/test_diff.py:190]: Instead of duplicating the `regenerate_test_data` code, one could use `@pytest.mark.parametrize` here.

I would guess something like
```python
@pytest.mark.parametrize("kwargs", [{'label': 'diff-relative'}, {...}])
def test_regenerate_test_data_relative(kwargs, mock_code_factory, generate_diff_inputs, datadir)
  mock_code = mock_code_factory(
      **kwargs,
      data_dir_abspath=...
```

**Review by ltalirz — APPROVED** [review]
Great, thank you @janssenhenning !

I guess one could still reduce code duplication in the tests a bit, but that could also be done in a later PR.

**Pull request merged**
